### ISSUE TITLE
Add admin column in post list table to display/sort unlisted posts.

### ISSUE THREAD

**Issue by Nikschavan** [Nikschavan/unlist-posts#25]

**Comment by matthewmcvickar** [comment]
I wanted to show posts that were unlisted in the admin post list and ended up solving this with a post filter section and label rather than a column.

The code below does the following:

1. Adds ' — Unlisted' next to unlisted posts in the post list, like you'd see next to a draft or private post.
2. Adds 'Unlisted' as a filter above the list of posts, similar to 'Mine' or 'Drafts,' et cetera.

```php
// Add 'Unlisted' next to the post title in the admin post list.
function add_unlisted_post_status_to_post_list( $states, $post ) {
  // Get unlisted posts.
  $unlisted_posts = maybe_unserialize( get_option( 'unlist_posts' ) );

  // Check if this post is unlisted.
  if ( in_array( $post->ID, $unlisted_posts ) ) {
    $states[] = 'Unlisted';
  }

  return $states;
}
add_filter( 'display_post_states', 'add_unlisted_post_status_to_post_list', 10, 2 );

// Add 'Unlisted' to filter list in the admin post list.
function admin_add_unlisted_post_filter( $views )  {
    // Get unlisted posts.
    $unlisted_posts = maybe_unserialize( get_option( 'unlist_posts' ) );
    $unlisted_count = count( $unlisted_posts );

    // Mark the link as current if this is the curret view.
    $link_attributes = '';
    if ( is_admin() && $_GET['post_status'] == 'unlisted' ) {
      $link_attributes = 'class="current" aria-current="page"';
    }

    $views['unlisted'] = '<a href="edit.php?post_status=unlisted&post_type=post" ' . $link_attributes . '>Unlisted <span class="count">(' . $unlisted_count . ')</span></a>';

    return $views;
}
add_filter( "views_edit-post", 'admin_add_unlisted_post_filter' );

// Parse the query for unlisted posts.
function admin_filter_unlisted_posts( $query ){
  global $pagenow;

  if (
    is_admin() && $pagenow == 'edit.php'
    && isset( $_GET['post_type'] ) && $_GET['post_type'] == 'post'
    && isset( $_GET['post_status'] ) && $_GET['post_status'] == 'unlisted'
  ) {
    $unlisted_posts = maybe_unserialize( get_option( 'unlist_posts' ) );
    $query->query_vars['post__in'] = $unlisted_posts;
  }

  return $query;
}
add_filter( 'parse_query', 'admin_filter_unlisted_posts' );
```

**Comment by Nikschavan** [comment]
Hi @matthewmcvickar 👋 

Thank you very much for this, Do you also want to try and send a PR with this change?

**Comment by matthewmcvickar** [comment]
@Nikschavan Sure! [Here's a PR.](https://github.com/Nikschavan/unlist-posts/pull/40) I couldn't get PHPCS to run, unfortunately; kept getting this error: `ERROR: Referenced sniff "PHPCompatibility" does not exist`. Maybe you can add contribution instructions to the README?

**Comment by Nikschavan** [comment]
This was released in v1.1.0, thank you for the PR @matthewmcvickar 👍
